libevm/VMFace.h

```diff
@@ -33,6 +33,13 @@ ETH_SIMPLE_EXCEPTION_VM(OutOfGas);
 ETH_SIMPLE_EXCEPTION_VM(OutOfStack);
 ETH_SIMPLE_EXCEPTION_VM(StackUnderflow);
 
+struct VmExecResult
+{
+	VmExecResult(owning_bytes_ref&& _output, bool _revertNeeded) : output(std::move(_output)), revertNeeded(_revertNeeded) {}
+
+	owning_bytes_ref output;
+	bool revertNeeded = false;
```

Comment:
That would be nice I think. I'm sure @gcolvin agrees :).
This is classic case "should a parser throw exception if syntax is invalid"? IMHO, not, because it is nothing exceptional about detecting syntex errors in parser. It is its _normal_ job. But that's more philosophical...

Anyway. Throw an exception in this PR. Refactoring of this should be separated.